column_buffer.go

```diff
@@ -184,36 +184,38 @@ func (col *optionalColumnBuffer) Pages() Pages {
 }
 
 func (col *optionalColumnBuffer) Page() BufferedPage {
-	numNulls := countLevelsNotEqual(col.definitionLevels, col.maxDefinitionLevel)
-	numValues := len(col.rows) - numNulls
+	if optionalRowsHaveBeenReordered(col.rows) {
```

Comment:
Make the fast case more straightforward? I don't know

```go
if !optionalRowsHaveBeenReordered(col.rows) {
     	return newOptionalPage(col.base.Page(), col.maxDefinitionLevel, col.definitionLevels)
} 
...
```

Reply:
Sure no problem, I was just replicating the same way the equivalent repeated buffer does it. Will do!